sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -56,7 +56,10 @@ class V2SessionCatalog(catalog: SessionCatalog, conf: SQLConf)
   override def listTables(namespace: Array[String]): Array[Identifier] = {
     namespace match {
       case Array(db) =>
-        catalog.listTables(db).map(ident => Identifier.of(Array(db), ident.table)).toArray
+        catalog
+          .listTables(db)
+          .map(ident => Identifier.of(Array(ident.database.getOrElse("")), ident.table))
```

Comment:
Matching the behavior or https://github.com/apache/spark/blob/850833fa177ec1f265e143fc383e40ec2c8341a6/sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala#L776